### PR TITLE
Add bin.demangle.flags to allow users to simplify or not mangled strings

### PR DESCRIPTION
**Pull request by wargio**
**Your checklist for this pull request**
- [ ] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [ ] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

...

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

**Review by XVilka — CHANGES_REQUESTED** [review]
```
In file included from ../librz/include/rz_vector.h:5,
                 from ../librz/include/rz_th.h:14,
                 from ../librz/include/rz_util.h:13,
                 from ../librz/demangler/demangler.c:6:
../librz/demangler/demangler.c: In function ‘rz_demangler_get_flags’:
../librz/include/rz_util/rz_assert.h:104:25: error: ‘return’ with no value, in function returning non-void [-Werror=return-type]
  104 |                         return; \
      |                         ^~~~~~
../librz/demangler/demangler.c:140:9: note: in expansion of macro ‘rz_return_if_fail’
  140 |         rz_return_if_fail(demangler);
      |         ^~~~~~~~~~~~~~~~~
../librz/demangler/demangler.c:139:24: note: declared here
  139 | RZ_API RzDemanglerFlag rz_demangler_get_flags(RZ_NONNULL RzDemangler *demangler) {
      |                        ^~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
```

**Review by XVilka — CHANGES_REQUESTED** [review]
Since it adds a new configuration variable, please up the project version with the empty conversion function.

File: librz/demangler/demangler.c [hunk 1]
```diff
@@ -41,43 +44,43 @@ RZ_LIB_VERSION(rz_demangler);
...
  * \brief Demangles rust symbols
  */
-RZ_API RZ_OWN char *rz_demangler_rust(RZ_NONNULL const char *symbol) {
-	return libdemangle_handler_rust(symbol);
+RZ_API RZ_OWN char *rz_demangler_rust(RZ_NONNULL const char *symbol, RzDemanglerFlag flags) {
+	return libdemangle_handler_rust(symbol, (RzDemangleOpts)flags);
 }
 
 /**
  * \brief Demangles microsft vc symbols
```
Comment (XVilka) [inline librz/demangler/demangler.c]: A typo

File: librz/demangler/demangler.c [hunk 2]
```diff
@@ -41,43 +44,43 @@ RZ_LIB_VERSION(rz_demangler);
...
  * \brief Demangles pascal symbols
  */
-RZ_API RZ_OWN char *rz_demangler_pascal(RZ_NONNULL const char *symbol) {
-	return libdemangle_handler_pascal(symbol);
+RZ_API RZ_OWN char *rz_demangler_pascal(RZ_NONNULL const char *symbol, RzDemanglerFlag flags) {
+	return libdemangle_handler_pascal(symbol, (RzDemangleOpts)flags);
 }
 
 /**
  * \brief Demangles rust symbols
```
Comment (XVilka) [inline librz/demangler/demangler.c]: "Rust"

File: librz/demangler/demangler.c [hunk 3]
```diff
@@ -41,43 +44,43 @@ RZ_LIB_VERSION(rz_demangler);
...
  * \brief Demangles objc symbols
  */
-RZ_API RZ_OWN char *rz_demangler_objc(RZ_NONNULL const char *symbol) {
-	return libdemangle_handler_objc(symbol);
+RZ_API RZ_OWN char *rz_demangler_objc(RZ_NONNULL const char *symbol, RzDemanglerFlag flags) {
+	return libdemangle_handler_objc(symbol, (RzDemangleOpts)flags);
 }
 
 /**
  * \brief Demangles pascal symbols
```
Comment (XVilka) [inline librz/demangler/demangler.c]: "Pascal"

File: librz/demangler/demangler.c [hunk 4]
```diff
@@ -41,43 +44,43 @@ RZ_LIB_VERSION(rz_demangler);
...
  * \brief Demangles c++ symbols
  */
-RZ_API RZ_OWN char *rz_demangler_cxx(RZ_NONNULL const char *symbol) {
-	return libdemangle_handler_cxx(symbol);
+RZ_API RZ_OWN char *rz_demangler_cxx(RZ_NONNULL const char *symbol, RzDemanglerFlag flags) {
+	return libdemangle_handler_cxx(symbol, (RzDemangleOpts)flags);
 }
 
 /**
  * \brief Demangles objc symbols
```
Comment (XVilka) [inline librz/demangler/demangler.c]: "ObjC"

File: librz/demangler/demangler.c [hunk 5]
```diff
@@ -41,43 +44,43 @@ RZ_LIB_VERSION(rz_demangler);
...
  * \brief Demangles java symbols
  */
-RZ_API RZ_OWN char *rz_demangler_java(RZ_NULLABLE const char *symbol) {
-	return libdemangle_handler_java(symbol);
+RZ_API RZ_OWN char *rz_demangler_java(RZ_NULLABLE const char *symbol, RzDemanglerFlag flags) {
+	return libdemangle_handler_java(symbol, (RzDemangleOpts)flags);
 }
 
 /**
  * \brief Demangles c++ symbols
```
Comment (XVilka) [inline librz/demangler/demangler.c]: "C++"

File: librz/demangler/demangler.c [hunk 6]
```diff
@@ -41,43 +44,43 @@ RZ_LIB_VERSION(rz_demangler);
 /**
  * \brief Demangles java symbols
```
Comment (XVilka) [inline librz/demangler/demangler.c]: "Java"

File: librz/demangler/demangler.c (hunk 1 above)
Comment (XVilka) [reply librz/demangler/demangler.c]: "Microsoft VC"

**Review by XVilka — CHANGES_REQUESTED** [review]

**Comment by wargio** [comment]
> Since it adds a new configuration variable, please up the project version with the empty conversion function.

There is no reason to do so. by default the output is the same as the old one.

**Comment by XVilka** [comment]
@wargio please fix those names and lets merge

File: librz/demangler/demangler.c [hunk 7]
```diff
@@ -39,45 +42,45 @@ static RzDemanglerPlugin *demangler_static_plugins[] = { RZ_DEMANGLER_STATIC_PLU
...
  */
-RZ_API RZ_OWN char *rz_demangler_rust(RZ_NONNULL const char *symbol) {
-	return libdemangle_handler_rust(symbol);
+RZ_API RZ_OWN char *rz_demangler_rust(RZ_NONNULL const char *symbol, RzDemanglerFlag flags) {
+	return libdemangle_handler_rust(symbol, (RzDemangleOpts)flags);
 }
 
 /**
- * \brief Demangles microsft vc symbols
+ * \brief Demangles Microsft VC symbols
```
Comment (XVilka) [inline librz/demangler/demangler.c]: Typo is still here

**Review by XVilka — APPROVED** [review]

File: librz/demangler/demangler.c (hunk 7 above)
Comment (wargio) [reply librz/demangler/demangler.c]: is not a typo :1st_place_medal:

**Review by XVilka — APPROVED** [review]

**Pull request merged**